src/nplinker/genomics/bgc.py

```diff
@@ -173,6 +174,36 @@ def is_mibig(self) -> bool:
         """
         return self.id.startswith("BGC")
 
+    def to_dict(self) -> dict[str, Any]:
```

Comment:
It's a nice idea to use `to_dict` to output a python class object to a structured and readable format, including but not limited to csv/tsv format. JSON is a another popular format for data exachage. So it's better to make sure the `to_dict` is JSON-compatiable, which means e.g. list should be used over tuple and set (https://docs.python.org/3.10/library/json.html#json.JSONDecoder). Dict could be used as the dict value, but note that the value of the inner dict should not e.g. use set or tuple neither.

Also, make sure that the value of the returned dict does not contain the special characters (e.g. comma, tab) since you want to output them to a csv/tsv.

Reply:
Instead of ensuring that the dict values do not contain special characters, I am replacing tabs in the to_tsv methods

Reply:
OK